charts/IndicatorStore.ts

```diff
@@ -0,0 +1,59 @@
+import { fetchJSON } from "./Util"
+import { Indicator } from "./Indicator"
+import { BAKED_BASE_URL } from "settings"
+
+export class IndicatorStore {
+    private indicatorsById: { [id: number]: Indicator } = {}
+    private fetchPromise?: Promise<void>
+
+    private async fetchAll() {
+        try {
+            const json = await fetchJSON(
+                `${BAKED_BASE_URL}/explore/indicators.json`
+            )
+            const indicators: Indicator[] = json.indicators
+            return indicators
+        } catch (error) {
+            console.error("Failed to fetch indicators", error)
+            return []
```

Comment:
If we fail to fetch, do we want to return an empty array, or trigger some sort of error?

Reply:
We should definitely let the user know, but before that we should also retry the request a few times, since there isn't much the user can do to fix it.

We don't have anything in the current codebase to show error dialogs or retry attempts, and this PR is a bit big as it is – maybe we should tackle that in another task?

Reply:
Yeah sure